src/gasPriceOracle/oracle.ts

```diff
@@ -14,29 +19,29 @@ export async function getGasPriceEstimate(
   chainId?: number,
   legacyFallback = true
 ): Promise<GasPriceEstimate> {
-  if (!chainId) {
-    const network: providers.Network = await provider.getNetwork();
-    chainId = network.chainId;
+  if (chainId === undefined) {
+    ({ chainId } = await provider.getNetwork());
   }
 
   const gasPriceFeeds: { [chainId: number]: GasPriceFeed } = {
-    1: eip1559,
-    10: eip1559,
-    137: polygonGasStation,
-    288: legacy,
-    324: legacy,
-    8453: eip1559,
-    42161: eip1559_arbitrum,
+    [CHAIN_IDs.ARBITRUM]: arbitrum.eip1559,
+    [CHAIN_IDs.BASE]: ethereum.eip1559,
+    [CHAIN_IDs.BOBA]: ethereum.legacy,
+    [CHAIN_IDs.LINEA]: linea.eip1559, // @todo: Support linea_estimateGas in adapter.
+    [CHAIN_IDs.MAINNET]: ethereum.eip1559,
+    [CHAIN_IDs.MODE]: ethereum.eip1559,
+    [CHAIN_IDs.OPTIMISM]: ethereum.eip1559,
+    [CHAIN_IDs.POLYGON]: polygon.gasStation,
+    [CHAIN_IDs.ZK_SYNC]: ethereum.legacy,
   };
 
-  let gasPriceFeed: GasPriceFeed = gasPriceFeeds[chainId];
+  let gasPriceFeed = gasPriceFeeds[chainId];
   if (gasPriceFeed === undefined) {
     if (!legacyFallback) {
       throw new Error(`No suitable gas price oracle for Chain ID ${chainId}`);
     }
-    gasPriceFeed = legacy;
+    gasPriceFeed = chainIsOPStack(chainId) ? ethereum.eip1559 : ethereum.legacy;
```

Comment:
We'll need to amend this later when other stacks are added, but for now it'll work fine for OP stack.